crates/frontend/src/pages/DefaultConfig/DefaultConfig.rs

```diff
@@ -184,23 +242,117 @@ pub fn DefaultConfig() -> impl IntoView {
+                                let new_key = filtered_key.split(".").map(str::to_string).collect::<Vec<String>>();
+                                if new_key.len() == 1 {
+                                    // key
+                                    ele.insert("key".to_string(), json!(new_key.get(0).unwrap().replace("\"", "")));
+                                } else {
+                                    // folder
+                                    let key = new_key.get(0).unwrap().to_owned().replace("\"", "")+".";
+                                    if !groups.contains(&key) {
+                                        ele.insert("key".to_string(), json!(key));
+                                        ele.insert("schema".to_string(), json!("-".to_string()));
+                                        ele.insert("value".to_string(), json!("-".to_string()));
+                                        ele.insert("function_name".to_string(), json!("-".to_string()));
+                                        ele.insert("created_at".to_string(), json!("-".to_string()));
+                                        ele.insert("created_by".to_string(), json!("-".to_string()));
+                                        groups.insert(key);
+                                    } else {
+                                        return None;
+                                    }
+                                }
+                                Some(ele)
+                            } else {
+                                None
+                            }
+                        }).collect::<Vec<Option<Map<String, Value>>>>();
+                        filtered_rows = opt_filtered_rows.into_iter().filter_map(|ele| ele).collect::<Vec<Map<String, Value>>>();
+                    }
+
+                    let total_default_config_keys = filtered_rows.len().to_string();
+
                     view! {
                         <div class="pb-4">
                             <Stat heading="Config Keys" icon="ri-tools-line" number=total_default_config_keys/>
                         </div>
                         <div class="card rounded-lg w-full bg-base-100 shadow">
                             <div class="card-body">
-                                <div class="flex justify-between">
-                                    <h2 class="card-title chat-bubble text-gray-800 dark:text-white bg-white font-mono">
-                                        "Default Config"
-                                    </h2>
-                                    <DrawerBtn drawer_id="default_config_drawer".to_string()>
-                                        Create Key <i class="ri-edit-2-line ml-2"></i>
-                                    </DrawerBtn>
+                                <div class="flex justify-between pb-2">
+                                    <div class="flex justify-between pt-3">{
+                                        {move || {
```

Comment:
```suggestion
```suggestion
                                        {
                                            move || {
                                                let data = bread_crums.get();
                                                let last_index = data.len() - 1;
                                                data.iter()
                                                    .enumerate()
                                                    .map(|(index, item)| {
                                                        let value = item.value.clone();
                                                        let is_link = item.is_link.clone();
                                                        let is_last_item = index < last_index;
                                                        let item_classname = if item.is_link {
                                                            "cursor-pointer text-blue-500 underline underline-offset-2"
                                                        } else {
                                                            ""
                                                        };
                                                        view! {
                                                            <div class="flex">
                                                                <h2
                                                                    on:click=move |_| {
                                                                        if is_link {
                                                                            folder_click_handler(value.clone())
                                                                        }
                                                                    }
                                                                    class={item_classname}
                                                                >
                                                                    {item.key.clone()}
                                                                </h2>
                                                                <h2 class="pl-4 pr-4">
                                                                    {if is_last_item { ">" } else { "" }}
                                                                </h2>
                                                            </div>
                                                        }
                                                    })
                                                    .collect_view()
                                            }
                                        }


```
```

Reply:
We can use derive signal like this, to solution this out

```
    let bread_crums = Signal::derive(move || {
        let filters = filters.get();
        let mut default_bread_crums = vec![BreadCrums {
            key: "Default Config".to_string(),
            value: None,
            is_link: true,
        }];

        let mut bread_crums = match filters {
            Some(prefix) => {
                let partials = prefix
                    .trim_matches('.')
                    .split(".")
                    .map(str::to_string)
                    .collect::<Vec<String>>();
                partials
                    .into_iter()
                    .fold(String::new(), |mut prefix, partial| {
                        default_bread_crums.push(BreadCrums {
                            key: partial.clone(),
                            value: Some(prefix.clone()),
                            is_link: true,
                        });
                        prefix.push_str(&partial);
                        prefix.push('.');
                        prefix
                    });
                default_bread_crums
            }
            None => default_bread_crums,
        };

        if let Some(last_crum) = bread_crums.last_mut() {
            last_crum.is_link = false;
        }

        bread_crums
    });

```